### DISCUSSION TITLE
Training from scratch: model architecture (e.g. yolo5l.yaml) specifies number of classes, but so does the --data flag yaml file. Should I change the model architecture yaml file to match the --data flag?

### DISCUSSION THREAD

**Discussion by Jakob-98** [ultralytics/yolov5#7974] [Q&A] [answered]
Training from scratch: model architecture (e.g. yolo5l.yaml) specifies number of classes, but so does the --data flag yaml file. Should I change the model architecture yaml file to match the --data flag?

E.g. my --data file specifies: 
`nc: 7
names: ['empty', 'human', 'fox', 'skunk', 'rodent', 'bird', 'other']
`

For training a model from scratch with 

`
python train.py --data custom.yaml --weights '' --cfg yolov5l.yaml
`

Should I create a yolov5l7c.yaml file specifying the nc=7 in that file to match the nc of my data file? This is not completely clear in https://github.com/ultralytics/yolov5/wiki/Tips-for-Best-Training-Results

Best,
Jakob

**Comment by glenn-jocher** [comment, marked as answer]
@Jakob-98 data.yaml nc overrides model.yaml nc if both are present.

**Reply by Jakob-98** [reply]
Thank you for your swift reply. Good to know! 

I hadn't encountered it in the docs yet.
